### ISSUE TITLE
support generating schema code?

### ISSUE THREAD

**Issue by teloon** [alpox/graphql-go-gen#2]
It seems currently it only parse the string source of a schema to GraphQLSchema in runtime. However, many times, we may want to change the generated schema a bit, e.g. add some resolve function.

This will be very useful in case like: get the schema from other places and ported it to go.

**Comment by alpox** [comment]
You're totally right! This project is still under development and is not stable yet and not all features added. I will go on working on it when i find the time but sadly my time is quite limited at the moment.
Many thanks for your response and interest!

**Comment by alpox** [comment]
For now, I added the possibility to extend the generated context before adding it to a schema, so one can add additional functions like Resolve functions.
Its not pretty yet and i want to provide a better usable interface for the future, but this is the state for now.

You find an Example for its usage in this test: https://github.com/alpox/graphql-go-gen/blob/master/generator/api_test.go
I hope that it helps! :-)

**Comment by teloon** [comment]
Or can we just generate the go code from the raw schema string? We can just edit it as we want.

**Comment by alpox** [comment]
There is a problem with that due to the design of graphql-go where the actual fields are hold private and object.Fields() returns not the actual Field collection but a different representation of the fields made for schema inspection.
The only workaround i found was to overwrite the field configuration which gets passed to an objects constructor afterwards, so the Extend function does it work before creating the actual schema.

The Extend function can be used on any type of object and gets its generated configuration as input, you can update that config and after returning it it will be updated in the context which you can pass to the schema creation function afterwards. That's maybe what you asked for now?

Im open for any ideas!
